src/mds/Server.cc

```diff
@@ -237,6 +237,9 @@ void Server::dispatch(Message *m)
     if (m->get_type() == MSG_MDS_SLAVE_REQUEST) {
       // handle_slave_request() will wait if necessary
       wait_for_active = false;
+    } else if (mds->is_stopping() && m->get_type() == CEPH_MSG_CLIENT_SESSION &&
+	       (static_cast<MClientSession*>(m))->get_op() != CEPH_SESSION_REQUEST_OPEN) {
+      wait_for_active = false;
```

Comment:
I'd prefer to see this as:

````
} else if (mds->is_stopping() && m->get_type() == CEPH_MSG_CLIENT_SESSION) {
  wait_for_active = false;
````

Move the check for `CEPH_SESSION_REQUEST_OPEN` to `handle_client_session` and reply with `MClientSession(CEPH_SESSION_REJECT)`?

Also, should we reject requests for `CEPH_SESSION_REQUEST_FLUSH_MDLOG`?

Reply:
CEPH_SESSION_REJECT may cause syscall to return error. I think it's better to just drop open request silently.